pkg/eventprocessor/impl/executor/impl/blockscope.go

```diff
@@ -162,6 +165,54 @@ func (bs *blockScope) TxnReceiptExists(ctx context.Context, txnHash common.Hash)
 	return true, nil
 }
 
+func (bs *blockScope) StateHash(ctx context.Context, chainID tableland.ChainID) (executor.StateHash, error) {
+	hash, err := dbhash.DatabaseStateHash(ctx, bs.txn, []dbhash.Option{
+		dbhash.WithFetchSchemasQuery(
```

Comment:
This query fetches all tables for a particular chain and all system tables that are relevant to state hash calculation